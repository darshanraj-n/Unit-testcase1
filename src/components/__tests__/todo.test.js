import {render,screen,cleanup} from '@testing-library/react'
import Todo from '../todo'
import renderer from 'react-test-renderer'

afterEach(()=>{
    cleanup()
})

test('should render incomplete todo component',()=>{
    const todo = {id: 1, title:'title 1', completed:false,}
    render(<Todo todo={todo}/>)
    const todoElement=screen.getByTestId('todo-1');
    expect(todoElement).toBeInTheDocument()
    expect(todoElement).toHaveTextContent('title 1')
    expect(todoElement).not.toContainHTML('<strike>');
})

test('should render todo component',()=>{
    const todo = {id: 2, title:'title 2', completed:true,}
    render(<Todo todo={todo}/>)
    const todoElement=screen.getByTestId('todo-2');
    expect(todoElement).toBeInTheDocument()
    expect(todoElement).toHaveTextContent('title 2')
})

test('matces snapshots',()=>{
    const todo = {id: 1, title:'title 1', completed:false,}
    const tree = renderer.create(<Todo todo={todo}/>).toJSON()
    expect(tree).toMatchSnapshot()
})