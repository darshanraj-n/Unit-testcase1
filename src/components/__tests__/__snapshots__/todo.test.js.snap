// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matces snapshots 1`] = `
<div
  data-testid="todo-1"
>
  <h1>
    title 1
  </h1>
</div>
`;
